event_reporter/utils/app_revision.go

```diff
@@ -46,6 +63,64 @@ func GetOperationRevision(a *appv1.Application) string {
 	return revision
 }
 
+func GetOperationSyncRevisions(a *appv1.Application) []string {
+	var revisions []string
+	if a != nil {
+		// this value will be used in case if application hasn't resources, like empty gitsource
+		revisions = getRevisions(RevisionsData{
+			Revision:  a.Status.Sync.Revision,
+			Revisions: a.Status.Sync.Revisions,
+		})
+
+		if a.Status.OperationState != nil && a.Status.OperationState.Operation.Sync != nil {
+			revisions = getRevisions(RevisionsData{
+				Revision:  a.Status.OperationState.Operation.Sync.Revision,
+				Revisions: a.Status.OperationState.Operation.Sync.Revisions,
+			})
+		} else if a.Operation != nil && a.Operation.Sync != nil {
+			revisions = getRevisions(RevisionsData{
+				Revision:  a.Operation.Sync.Revision,
+				Revisions: a.Operation.Sync.Revisions,
+			})
+		}
+	}
+
+	return revisions
+}
+
+// for monorepo support: list with revisions where actual changes to source directory were committed
+func GetOperationChangeRevisions(a *appv1.Application) []string {
+	var revisions []string
+	if a != nil {
+		// this value will be used in case if application hasn't resources, like empty gitsource
+		if a.Status.OperationState != nil && a.Status.OperationState.Operation.Sync != nil {
+			if a.Status.OperationState.Operation.Sync.ChangeRevision != "" || a.Status.OperationState.Operation.Sync.ChangeRevisions != nil {
+				revisions = getRevisions(RevisionsData{
+					Revision:  a.Status.OperationState.Operation.Sync.ChangeRevision,
+					Revisions: a.Status.OperationState.Operation.Sync.ChangeRevisions,
+				})
+			}
+		} else if a.Operation != nil && a.Operation.Sync != nil {
```

Comment:
I think we will rid of idea to put it on operation level, we will just persist it on status

Reply:
it's just as a fallback so can leave ATM